term_proposal/titaneric/proposal.md

```diff
@@ -0,0 +1,52 @@
+# Basic neural network library supported auto-differentiation
+
+## Motivation
+
+Most deep learning courses aim to teach math behind the network, architecture and their applications.
+
+However, seldom course talk about how to implement the deep learning library and start every from scratch.
+
+Wish to implement this kind of library and learn how to design a better numerical software.
+
+## Target
+
+Based on `autograd` project, build a similar library that user simply define the function, and this lib can automatically calculate this differentiation form of given function.
+
+Build the computational graph when function is called, calculate the backward propogation with respect to variable.
+
+Provide a benchmark compared to tensorflow and pytorch.
+
+If time allowed, provide a simple multi-layer perceptron (neural network) interface, criterion, optimizer, datasets and dataloader like pytorch.
+
+
+## Goal Feature
```

Comment:
It's not clear to me what is the difference between the section Target and Goal Feature.  Perhaps you can specify more details in this section, or combine it in the previous one.